crates/red_knot/src/types.rs

```diff
@@ -46,6 +51,45 @@ impl Type {
     pub const fn is_unknown(&self) -> bool {
         matches!(self, Type::Unknown)
     }
+
+    // NOTE: naming? `get_member_type`
+    pub fn get_member(&self, db: &dyn SemanticDb, name: &Name) -> QueryResult<Type> {
+        // NOTE: maybe make this a trait that each TypeID implements?
```

Comment:
I don't think we necessarily need a trait. Defining a `type_id.get_member` method seems sufficient because we don't need to dynamically call `get_member` on any `type_id` (we use static dispatch here)